plugins/destination/s3/client/client.go

```diff
@@ -63,17 +67,75 @@ func New(ctx context.Context, logger zerolog.Logger, spec specs.Destination) (de
 	c.downloader = manager.NewDownloader(c.s3Client)
 
 	// we want to run this test because we want it to fail early if the bucket is not accessible
-	timeNow := time.Now().UTC()
-	if _, err := c.uploader.Upload(ctx, &s3.PutObjectInput{
-		Bucket: aws.String(c.pluginSpec.Bucket),
-		Key:    aws.String(replacePathVariables(c.pluginSpec.Path, "TEST_TABLE", "TEST_UUID", timeNow)),
-		Body:   bytes.NewReader([]byte("")),
-	}); err != nil {
-		return nil, fmt.Errorf("failed to write test file to S3: %w", err)
+	if err := c.testWriteAccess(ctx, cfg); err != nil {
+		return nil, err
 	}
+
 	return c, nil
 }
 
 func (*Client) Close(ctx context.Context) error {
 	return nil
 }
+
+func (c *Client) testWriteAccess(ctx context.Context, cfg aws.Config) error {
```

Comment:
So this only works if the bucket and user credentials are in the same account. I think we are better off just adding an option that allows users to skip the tests